### PR TITLE
introduce auto_mp4 request flag

### PR DESCRIPTION
**Pull request by mjh1**
only generate mp4s if auto_mp4 request flag is set and input duration is less than 2mins

Fixes https://github.com/livepeer/catalyst-api/issues/418

### CODE REVIEW

**Comment by emranemran** [comment]
Changes LGTM. @victorges and @gioelecerati: mind reviewing when you get a chance? We'll also hold off on the corresponding studio changes before merging.

**Review by gioelecerati — APPROVED** [review]
LGTM, 
this is the commit on the task-runner side if you want to take a look https://github.com/livepeer/task-runner/pull/132/commits/bef71b859355ad30469cffefd2490f773081bce6

File: handlers/upload.go
Comment (victorges) [inline handlers/upload.go:93]: WDYT of returning the location of the `URL` of the MP4 location instead and saving it on the args object? Otherwise we're completely ignoring it here and assuming it has to be the same location as the HLS playlist etc.

If it requires too much changes in the core logic (like splitting the logic to copy the files etc which might be cumbersome), another idea would be returning this "is auto mp4" boolean from the `getTargetURL` function instead. So we explicitly only support the same output location for both locations, and just return whether the "target url" location has `auto_mp4` turned on.

**Review by victorges — APPROVED** [review]
LGTM either way

File: handlers/upload.go
Comment (mjh1) [reply handlers/upload.go:93]: yeah i was considering doing the latter tbh, since we do currently only support a single output location. i'll go with that

File: handlers/upload.go
Comment (victorges) [inline handlers/upload.go:84]: Nice! Me gusta :)

**Review by victorges — APPROVED** [review]
LGTM

**Pull request merged**
